include/odp/api/spec/traffic_mngr.h

```diff
@@ -244,6 +244,10 @@ typedef struct {
 	 * When true the min_weight and max_weight fields above specify
 	 * the legal range of such weights. */
 	odp_bool_t weights_supported;
+
+	/** tm_node_threshold_supported indicates that the tm_nodes at this
+	 * level support threshold profile set. */
+	odp_bool_t tm_node_threshold_supported;
```

Comment:
Why do we need this capability field instead of requiring the threshold being always supported? As far as I can see, the threshold profile is basically just maximum queue length specification and thus seems mandatory everywhere where there can be queues.

That said, I do not quite understand why there is threshold config (and wred config) in every level as opposed to only in tm_queues. If tm_queues are the only place packets get queues, one would think that that is the only place where wred or tail drop gets applied so it is hard to see what the queue related parameters for tm nodes do. If there is some sort of virtual queuing in may levels so that the multilevel threshold and wred config would make sense, I think the API fails to describe it.

If this capability field makes sense and ends up getting added in the API, then I think corresponding field is needed in the level requirements struct.

I think we could drop _supported suffix from the newly added capability fields. It is a bit redundant since capability structures are about what is supported and what is not. This comment applies to all the added fields in this patch set.

Reply:
In our HW, we don't support WRED and we cannot support tail drop based on max_bytes on leaf/SQ.
On TM nodes' we don't support any kind of thresholds and as you said, I'm not sure if there are any HW out which support such behavior hence added it in capabilities.

I'll fix the name and add it in requirements struct as well.